code/obj/machinery/computer/syndicate/podbay.dm

```diff
@@ -0,0 +1,124 @@
+// Riot computer but reflavored for the Cairngorm
+
+/obj/machinery/computer/battlecruiser_podbay
+	name = "podbay authorization"
+	icon_state = "drawbr"
+	density = FALSE
+	glow_in_dark_screen = TRUE
+	/// How many authentications are needed to release the shielding on the podbay. Set the first time someone clicks this.
+	var/auth_need = null
+	var/list/authorized
+	desc = "Controls access to the podbay. Use this when your team is ready to go."
+
+	light_r = 1
+	light_g = 0.3
+	light_b = 0.3
+
+	/// Whether or not the podbay on the Cairngorm is authorized.
+	var/authed = FALSE
+	/// How long until we take matters into our own hands.
+	var/auth_delay = 10 MINUTES
+	var/area/battlecruiser_area
+
+	New()
+		..()
+		START_TRACKING
+
+	initialize()
+		battlecruiser_area = get_area_by_type(/area/syndicate_station/battlecruiser)
+		SPAWN(auth_delay)
+			authorize() // If they haven't done it before auth_delay, do it for em
+		..()
+
+	disposing()
+		src.authorized = null
```

Comment:
I admit that I am not entirely familiar with the full process `qdel()` and garbage collection, but I believe any variables held by the deleted datum should be nulled, making `src.authorized = null` superfluous.

The verdict of someone with greater knowledge of BYOND's garbage collection would be appreciated.

Reply:
From what I understand in imcoder discussion, setting the list to null (and its len to 0 too) is a good idea.

Reply:
Or not actually I can probably remove this. Qdel should handle it